.github/workflows/pr-image.yaml

```diff
@@ -0,0 +1,32 @@
+name: Build and push image from PR
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build:
+    name: Build and push image from PR
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

Comment:
Can we update this to `v4`?  `v3` is deprecated.  May also be a good idea to do a drive-by update on `build.yaml` too please.